content/edpratti/collections/git-guide/posts/git-into-the-weeds/index.md

```diff
@@ -0,0 +1,240 @@
+
+> **This is classified as a `modify/delete` conflict.:**
+> To resolve it, you must choose to either re-add the file or confirm its deletion.
+
+### Two branches containing different contents for the same line.
+
+<img src="./git_conflict_modified.svg" alt="A commit history showing branches modifying the same lines, resulting in a git merge conflict." style="background-color: var(--background_primary-bright); border: var(--border-width_s) solid var(--background_disabled); border-radius: var(--corner-radius_xl)">
+
+> **This is classified as a `content` conflict.:**
+> Git will insert conflict markers within the file as opposed to simply showing a conflict.
+
+**There are also other types of conflicts, like the following:**
+
+| Output | Explanation |
+| --- | --- |
+| `(add/add)` | Different files are added under the same filename in different branches. To resolve this, simply rename one of the files manually and attempt to merge again. |
+| `(rename/rename)` | Same file with two different filenames in different branches. To resolve this, rename one of the files manually and attempt to merge again. |
+| `(rename/modify)` | The same file is renamed in one branch and modified in another under its old filename. Git will provide two files; the original and a `renamed` file. Either apply the changes to the `renamed` file or delete the original and attempt to merge again. |
+
+### On a Git GUI
+
+On a Git client, this is much simpler. Git GUIs will allow you to select which modification to pick as the resolution for the conflict.
+
+<img src="./git_conflict_gui.svg" alt="An example of a hypothetical Git conflict GUI. It shows options to choose which version of a file to select, one that's been deleted or one that's been modified." style="background-color: var(--background_primary-bright); border: var(--border-width_s) solid var(--background_disabled); border-radius: var(--corner-radius_xl)">
+
+In most Git clients you'll get a side-by-side view and the option to pick which of the versions of the file or files you want to keep. This makes managing conflicts much easier for the user.
+
+---
+
+# Submodules
+
+Submodules are repositories within repositories. They allow you to use existing libraries and dependencies within your projects and can be updated independently.
+
+## How to add a submodule
+
+We can add a submodule with the following syntax:
+
+```bash
+git submodule add [repository-url] [path]
+```
+
+Now let's try with an example, like a design library:
+
+```bash
+git submodule add https://github.com/mui/material-ui.git libs/material-ui
+git commit -m "Add MaterialUI as a submodule"
+```
+
+> **The repository is then cloned in the specified path.:**
+
+<!-- ::start:filetree -->
+- `docs/`
+- `libs/`
+  - `material-ui/`
```

Comment:
Can we highlight this directory using the syntax mentioned in `FEATURES.md` on the repo?